Graphs/Astar.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * @author : Mathang Peddi
+ * A* Algorithm calculates the minimum cost path between two nodes.
+ * It is used to find the shortest path using heuristics.
+ * It uses graph data structure.
+ */
+
+// Euclidean distance heuristic for 2D points
+function euclideanHeuristic(pointA, pointB) {
+  const dx = pointA[0] - pointB[0]
+  const dy = pointA[1] - pointB[1]
+  return Math.sqrt(dx * dx + dy * dy)
+}
+
+// Priority Queue (Min-Heap) implementation
+class PriorityQueue {
```

Comment:
Please import this (we have a priority queue implementation in this repo) rather than reimplementing it.